integration_tests/deploy_check.py

```diff
@@ -103,5 +98,18 @@ def _deploy_and_test(appdir, language, is_xrt):
             deploy_app.stop_app(version)
 
 
+@retry(wait_fixed=4000, stop_max_attempt_number=8)
+def _test_application(application_url):
+    output, status_code = test_util.get(application_url)
+
+    if status_code:
+        message = 'Application returned non-zero status code: %d' % status_code
+        logging.error(message)
```

Comment:
let's remove the `logging.error()` call here (since it will get printed to stderr by raising the exception anyway. then you can just do `raise RuntimeError('Application .... %d', status_code)` (note the comma instead of a `%`)